server/backend/neighbors.py

```diff
@@ -303,6 +303,15 @@ def query_model_for_tok(
     # Check to see if token is in the vocab
     vocab = set(word_vectors.key_to_index.keys())
 
+    if "mesh" in tok:
```

Comment:
Minor point, but we might end up changing tokens that just include the substring "mesh" if we use `"mesh" in tok` as the predicate. What about something like the below instead? I've also included some debugging output and reformatted the if-else blocks a bit -- feel free to keep or discard whatever you want from this.

```suggestion
    if tok.startswith("mesh_"):
        original_tok = tok
        tok = (
            f"disease_{tok}" if f"disease_{tok}" in vocab
            else f"chemical_{tok}" if f"chemical_{tok}" in vocab
            else tok
        )

        if tok != original_tok:
            logger.info(f"remapped token {original_tok} to {tok}")

```